File: java/spring-boot/user/src/main/java/com/learnwy/login/controller/LoginController.java
package com.learnwy.login.controller;

import com.learnwy.UserMain;
import com.learnwy.response.Message;
import com.learnwy.user.modal.UserModal;
import com.learnwy.user.service.UserService;
import com.sun.org.apache.xpath.internal.operations.Bool;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

/**
 * date 2017/9/4
 * create by learnwy
 * lis is GPL-3.0
 */
@RestController
@RequestMapping("/login")
public class LoginController {
    @Autowired
    UserService userService;

    @RequestMapping(path = {""}, method = {RequestMethod.POST})
    public Message login(@RequestBody Map<String, String> nameAndPassword) {
        List<UserModal> users = userService.findByUserLoginNameAndUserPassword(nameAndPassword.get("userName"), nameAndPassword.get("password"));
        if (users.size() == 0) {
            return Message.successMessage("登录失败:用户名或密码错误");
        }
        return Message.successMessage("登录成功");
    }
}

File: kotlin/kotlinSpringBoot/src/main/kotlin/com/learnwy/kotlinSpringBoot/system/controllers/LoginController.kt
package com.learnwy.kotlinSpringBoot.system.controllers

import com.learnwy.kotlinSpringBoot.Request
import com.learnwy.kotlinSpringBoot.system.data.User
import com.learnwy.kotlinSpringBoot.system.services.LoginService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.servlet.view.RedirectView
import javax.servlet.http.HttpSession
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


@RestController
@RequestMapping(path = arrayOf("/login"))
class LoginController {
    companion object {
        var logger: Logger = LoggerFactory.getLogger(LoginController::class.java);
    }

    @Autowired
    lateinit var loginService: LoginService;

    @RequestMapping(path = arrayOf(), method = arrayOf(RequestMethod.POST))
    fun login(httpSession: HttpSession, @RequestBody req: Request<User>) {
        if (loginService.login(req.data, httpSession)) {
//            RedirectView("/index")
        } else {
//            RedirectView("/login");
        };
    }

    @RequestMapping(path = arrayOf(), method = arrayOf(RequestMethod.GET))
    fun loginView(httpSession: HttpSession): String {
        return "login";
    }
}